ERROR LOG: Camunda Exporter fails to retrieve mappings for indices/templates

## Description

Several Errors have been reported on SaaS for the Alpha1 (MVP) that indices haven't been found, this was logged as error.

### Severity:

 * Low: As check will be retried, other partitions are creating the indices in the meantime and succeed.

### Impact

 * Low: As retry is possible and done. It is just noise.

### Stacktrace   

```
io.camunda.exporter.exceptions.ElasticsearchExporterException: Failed retrieving mappings from index/index templates with pattern [identity-users-8.7.0_*,identity-authorizations-8.7.0_*,tasklist-form-8.4.0_*,tasklist-metric-8.3.0_*,operate-process-8.3.0_*,operate-metric-8.3.0_*,operate-decision-requirements-8.3.0_*,operate-decision-8.3.0_*]
	at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:131) ~[camunda-exporter-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.exporter.schema.SchemaManager.initialiseIndices(SchemaManager.java:72) ~[camunda-exporter-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.exporter.schema.SchemaManager.initialiseResources(SchemaManager.java:62) ~[camunda-exporter-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.exporter.schema.SchemaManager.startup(SchemaManager.java:49) ~[camunda-exporter-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.exporter.CamundaExporter.open(CamundaExporter.java:104) ~[camunda-exporter-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.broker.exporter.stream.ExporterContainer.lambda$openExporter$0(ExporterContainer.java:118) ~[zeebe-broker-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.util.jar.ThreadContextUtil.runCheckedWithClassLoader(ThreadContextUtil.java:59) ~[zeebe-util-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.util.jar.ThreadContextUtil.runWithClassLoader(ThreadContextUtil.java:35) ~[zeebe-util-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.broker.exporter.stream.ExporterContainer.openExporter(ExporterContainer.java:117) ~[zeebe-broker-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirector.lambda$startActiveExportingMode$16(ExporterDirector.java:532) ~[zeebe-broker-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.retry.BackOffRetryStrategy.run(BackOffRetryStrategy.java:51) ~[zeebe-scheduler-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:83) [zeebe-scheduler-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:42) [zeebe-scheduler-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:122) [zeebe-scheduler-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130) [zeebe-scheduler-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108) [zeebe-scheduler-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227) [zeebe-scheduler-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
Caused by: co.elastic.clients.elasticsearch._types.ElasticsearchException: [es/indices.get_mapping] failed: [index_not_found_exception] no such index [identity-users-8.7.0_]
	at co.elastic.clients.transport.ElasticsearchTransportBase.getApiResponse(ElasticsearchTransportBase.java:345) ~[elasticsearch-java-8.13.4.jar:?]
	at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:147) ~[elasticsearch-java-8.13.4.jar:?]
	at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1210) ~[elasticsearch-java-8.13.4.jar:?]
	at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.getMapping(ElasticsearchIndicesClient.java:1226) ~[elasticsearch-java-8.13.4.jar:?]
	at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getCurrentMappings(ElasticsearchEngineClient.java:199) ~[camunda-exporter-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	at io.camunda.exporter.schema.elasticsearch.ElasticsearchEngineClient.getMappings(ElasticsearchEngineClient.java:114) ~[camunda-exporter-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
	... 16 more
```

### Occurrences:

* https://console.cloud.google.com/errors/detail/COmkg86i-5apRA;service=zeebe;time=P7D;locations=global?project=camunda-saas-int
* https://console.cloud.google.com/errors/detail/CNrntbKDxvPaLQ;service=zeebe;time=P7D;locations=global?project=camunda-saas-int
*

--- a/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/elasticsearch/ElasticsearchEngineClient.java
+++ b/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/elasticsearch/ElasticsearchEngineClient.java
@@ -196,7 +196,12 @@ public class ElasticsearchEngineClient implements SearchEngineClient {
   private Map<String, TypeMapping> getCurrentMappings(
       final MappingSource mappingSource, final String namePattern) throws IOException {
     if (mappingSource == MappingSource.INDEX) {
-      return client.indices().getMapping(req -> req.index(namePattern)).result().entrySet().stream()
+      return client
+          .indices()
+          .getMapping(req -> req.index(namePattern).ignoreUnavailable(true))
+          .result()
+          .entrySet()
+          .stream()
           .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().mappings()));
     } else if (mappingSource == MappingSource.INDEX_TEMPLATE) {
       return client

--- a/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/opensearch/OpensearchEngineClient.java
+++ b/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/opensearch/OpensearchEngineClient.java
@@ -266,7 +266,12 @@ public class OpensearchEngineClient implements SearchEngineClient {
   private Map<String, TypeMapping> getCurrentMappings(
       final MappingSource mappingSource, final String namePattern) throws IOException {
     if (mappingSource == MappingSource.INDEX) {
-      return client.indices().getMapping(req -> req.index(namePattern)).result().entrySet().stream()
+      return client
+          .indices()
+          .getMapping(req -> req.index(namePattern).ignoreUnavailable(true))
+          .result()
+          .entrySet()
+          .stream()
           .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().mappings()));
     } else if (mappingSource == MappingSource.INDEX_TEMPLATE) {
       return client

--- a/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/schema/ElasticsearchEngineClientIT.java
+++ b/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/schema/ElasticsearchEngineClientIT.java
@@ -9,6 +9,7 @@ package io.camunda.exporter.schema;
 
 import static io.camunda.exporter.schema.SchemaTestUtil.validateMappings;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
 import static org.mockito.Mockito.*;
 
 import co.elastic.clients.elasticsearch.ElasticsearchClient;
@@ -142,6 +143,22 @@ public class ElasticsearchEngineClientIT {
                 .name("world")
                 .typeDefinition(Map.of("type", "keyword"))
                 .build());
+  }
+
+  @Test
+  void shouldNotThrowErrorIfRetrievingMappingsWhereOnlySubsetOfIndicesExist() {
+    // given
+    final var index =
+        SchemaTestUtil.mockIndex("index_qualified_name", "alias", "index_name", "/mappings.json");
+
+    elsEngineClient.createIndex(index, new IndexSettings());
+
+    // when, tnen
+    assertThatNoException()
+        .isThrownBy(
+            () ->
+                elsEngineClient.getMappings(
+                    index.getFullQualifiedName() + "*,foo*", MappingSource.INDEX));
   }
 
   @Test

--- a/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/schema/OpensearchEngineClientIT.java
+++ b/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/schema/OpensearchEngineClientIT.java
@@ -8,6 +8,7 @@
 package io.camunda.exporter.schema;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
@@ -228,6 +229,22 @@ public class OpensearchEngineClientIT {
                 .name("world")
                 .typeDefinition(Map.of("type", "keyword"))
                 .build());
+  }
+
+  @Test
+  void shouldNotThrowErrorIfRetrievingMappingsWhereOnlySubsetOfIndicesExist() {
+    // given
+    final var index =
+        SchemaTestUtil.mockIndex("index_qualified_name", "alias", "index_name", "/mappings.json");
+
+    opensearchEngineClient.createIndex(index, new IndexSettings());
+
+    // when, then
+    assertThatNoException()
+        .isThrownBy(
+            () ->
+                opensearchEngineClient.getMappings(
+                    index.getFullQualifiedName() + "*,foo*", MappingSource.INDEX));
   }
 
   @Test

